src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/FlowScanningUtils.java

```diff
@@ -113,13 +114,14 @@ public int compare(@CheckForNull FlowNode first, @CheckForNull FlowNode second)
      * Returns all {@link BlockStartNode}s enclosing the given FlowNode, starting from the inside out.
      * This is useful if we want to obtain information about its scope, such as the workspace, parallel branch, or label.
      * Warning: while this is efficient for one node, batch operations are far more efficient when handling many nodes.
+     * <p>Deprecated, prefer {@link FlowNode#iterateEnclosingBlocks()} instead, possibly using {@link FilteratorImpl#FilteratorImpl(Iterator, Predicate)}
```

Comment:
Consider prototyping such a change against [this](https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/044bbd653de6077b9d1b7871a618444253716647/pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy#L254) and/or [this](https://github.com/jenkinsci/workflow-durable-task-step-plugin/blob/ce319baddf659aca192aec673f53f40c3ee51755/src/main/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepExecution.java#L478).

Reply:
@jglick Actually, fetchEnclosingBlocks uses the new APIs now :)

Reply:
So basically you get the goodies without extra hassle.